### PR TITLE
Update versions for v5.0 release

### PR DESCRIPTION
**Pull request by deriamis**

### CODE REVIEW

File: Formula/libmongocrypt.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Libmongocrypt < Formula
```
Comment (acmorrow) [inline Formula/libmongocrypt.rb:1]: Is this a necessary change for the 5.0 GA, in that libmongocrypt 1.2.1 must be available in order for 5.0 bits to work? Or is it more of a drive-by? If the latter, I'd prefer we did it as a separate commit.

File: Formula/mongodb-community-shell.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ class MongodbCommunityShell < Formula
 
   # frozen_string_literal: true
 
-  url "https://fastdl.mongodb.org/osx/mongodb-shell-macos-x86_64-4.2.0.tgz"
-  sha256 "68416de0b9652187ad2aa95c97ba31a031094c270975b23bc1f4ca22ee633360"
+  url "https://fastdl.mongodb.org/osx/mongodb-shell-macos-x86_64-5.0.0.tgz"
```
Comment (acmorrow) [inline Formula/mongodb-community-shell.rb:7]: Apparently we never updated this for 4.4?

File: Formula/mongodb-community@4.4.rb [hunk 3]
```diff
@@ -1,14 +1,16 @@
-class MongodbCommunityAT32 < Formula
+class MongodbCommunityAT44 < Formula
```
Comment (acmorrow) [inline Formula/mongodb-community@4.4.rb:1]: Is this just git similarity detection being off in the weeds? What I would have expected to see here was that `Formula/mongodb-community.rb` was renamed to `Formula/mongodb-community@4.4.rb`.

File: Formula/mongodb-community@5.0.rb [hunk 4]
```diff
@@ -1,14 +1,16 @@
-class MongodbCommunityAT34 < Formula
+class MongodbCommunityAT50 < Formula
```
Comment (acmorrow) [inline Formula/mongodb-community@5.0.rb:1]: I don't think there should be a `Formula/mongodb-community@5.0.rb` if we are following the pattern from the 4.4 GA.  If we are doing that, Instead you need to make a symlink from `Aliases/mongodb-community@5.0.rb` to `Formula/mongodb-community@5.0.rb` and remove `Aliases/mongodb-community@4.4.rb`. I think something also needs to be done with `Aliases/mongodb-mongocryptd@4.4.rb`. Yes, it's weird.

**Review by acmorrow — CHANGES_REQUESTED** [review]

File: Formula/libmongocrypt.rb (hunk 1 above)
Comment (deriamis) [reply Formula/libmongocrypt.rb:1]: I don't really know, and I don't have a way of testing. I'll make it separate, though.

File: Formula/mongodb-community-shell.rb (hunk 2 above)
Comment (deriamis) [reply Formula/mongodb-community-shell.rb:7]: Apparently not!

File: Formula/mongodb-community@4.4.rb (hunk 3 above)
Comment (deriamis) [reply Formula/mongodb-community@4.4.rb:1]: It's git similarity detection. It detected that `v3.2` got renamed over `v4.4` for some inexplicable reason.

File: Formula/mongodb-community@5.0.rb (hunk 4 above)
Comment (deriamis) [reply Formula/mongodb-community@5.0.rb:1]: Our documentation says to install the `@5.0` version specifically, which is why I made it. There's a bit of nuance surrounding the class name and the target, so I don't think a symlink would work in this instance. I think it's better to just update both the non-targeted and targeted files going forward.

File: Formula/libmongocrypt.rb (hunk 1 above)
Comment (acmorrow) [reply Formula/libmongocrypt.rb:1]: I think #116 already does it. Maybe we should just merge that too? @kevinAlbs were you waiting on a particular reviewer on that?

File: Formula/mongodb-community@5.0.rb (hunk 4 above)
Comment (deriamis) [reply Formula/mongodb-community@5.0.rb:1]: Oh, yeah, I missed `mongocryptd` completely. I'll get that.

File: Formula/libmongocrypt.rb (hunk 1 above)
Comment (kevinAlbs) [reply Formula/libmongocrypt.rb:1]: #116 omitted the update to `BUILD_VERSION`. If it is all the same, we can close #116 and proceed with this. Otherwise I can update #116 and request a reviewer.

File: Formula/mongodb-community@5.0.rb (hunk 4 above)
Comment (acmorrow) [reply Formula/mongodb-community@5.0.rb:1]: Right, but the way it (appears to) work is that you want `mongodb-community` with no adornment to be the thing that installs 5.0. Then you make an Alias entry for that called @5.0. Have a look at how it is structured currently w.r.t. 4.4.

File: Formula/libmongocrypt.rb (hunk 1 above)
Comment (acmorrow) [reply Formula/libmongocrypt.rb:1]: OK. @deriamis I'd say keep your edits for mongocrypt, but please land this as two separate commits, one to update mongocrypt, and another for the 5.0 GA. Then close #116.

File: Formula/mongodb-community@5.0.rb (hunk 4 above)
Comment (acmorrow) [reply Formula/mongodb-community@5.0.rb:1]: Ditto re `Aliases/mongodb-mongocryptd@4.4` to `Aliases/mongodb-mongocryptd@5.0`

Comment (deriamis) [reply Formula/mongodb-community@5.0.rb]: Ah, right. I had forgotten how that worked for some reason. I think Daniel's last commit that we reverted added a new class rather than update the Aliases, which is why I got confused.

File: Formula/mongodb-mongocryptd.rb [hunk 5]
```diff
@@ -1,9 +1,9 @@
-class MongodbMongocryptd < Formula
+class MongodbMongocryptdAT50 < Formula
```
Comment (acmorrow) [inline Formula/mongodb-mongocryptd.rb:1]: I don't think you want the `AT50` here.

File: Formula/mongodb-mongocryptd.rb [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
   desc "mongocryptd service for Client Side Encryption"
   homepage "https://www.mongodb.com/"
 
-  url "https://downloads.mongodb.com/osx/mongodb-cryptd-macos-x86_64-enterprise-4.4.6.tgz"
-  sha256 "bdf8aa2110b1f72c847b6b49f73fb855ae58549a2640ea69d7971740d7ff10db"
+  url "https://downloads.mongodb.com/osx/mongodb-macos-x86_64-enterprise-5.0.0.tgz"
```
Comment (acmorrow) [inline Formula/mongodb-mongocryptd.rb:5]: I don't see `cryptd` in the filename here anymore.

File: Formula/mongodb-community@3.4.rb [hunk 7]
```diff
@@ -1,81 +0,0 @@
-class MongodbCommunityAT34 < Formula
```
Comment (acmorrow) [inline Formula/mongodb-community@3.4.rb:1]: Is the rationale for removing this that 3.4 is EOL? If so, should we remove 3.6 as well?

Comment (acmorrow) [reply Formula/mongodb-community@3.4.rb:1]: And, why aren't we removing 3.2 as well, if we are doing this? Do we actually need to fold in these removals to the 5.0 GA change? If not, I'd consider backing them out.

File: Formula/mongodb-community@4.4.rb (hunk 3 above)
Comment (acmorrow) [reply Formula/mongodb-community@4.4.rb:1]: Oh, I guess that you *did* remove the 3.2 Formula? So it thinks you renamed it. Let's just restore all the EOL versions here and do the removals in a follow up.

**Review by acmorrow — CHANGES_REQUESTED** [review]
I see you backed out the `libmongocrypt` update. That's fine, we will just proceed with doing that via #116.

File: Formula/mongodb-community@4.4.rb (hunk 3 above)
Comment (deriamis) [reply Formula/mongodb-community@4.4.rb:1]: Okay, sure.

File: Formula/mongodb-mongocryptd.rb (hunk 5 above)
Comment (deriamis) [reply Formula/mongodb-mongocryptd.rb:1]: I thought I'd removed that. Woops.

**Comment by deriamis** [comment]
> I see you backed out the `libmongocrypt` update. That's fine, we will just proceed with doing that via #116.

One of these times, I am going to read everything before I act. I still had everything in my stash, so I just reapplied it and added a new commit.

File: Formula/libmongocrypt.rb (hunk 1 above)
Comment (deriamis) [reply Formula/libmongocrypt.rb:1]: I still had this in my stash, so I checked it back out and added another commit to this PR.

**Review by acmorrow — APPROVED** [review]
This LGTM. Ship it! Please rebase above head of master rather than creating a merge commit.
